xrpl-wasm-std/src/core/types/issue.rs

```diff
@@ -1,66 +1,188 @@
+pub const NO_ACCOUNT: AccountID =
+    AccountID([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]);
+
+/// Issue represents a currency with an issuer (IOU).
+/// This mirrors the C++ Issue struct.
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 #[repr(C)]
-pub struct IouIssue {
-    issuer: AccountID,
-    currency: Currency,
-    _bytes: [u8; 40],
+pub struct Issue {
+    /// The currency code (20 bytes)
+    pub currency: Currency,
+    /// The issuer account ID (20 bytes)
+    pub account: AccountID,
 }
 
-impl IouIssue {
-    pub fn new(issuer: AccountID, currency: Currency) -> Self {
-        let mut bytes = [0u8; 40];
-        bytes[..20].copy_from_slice(currency.as_bytes());
-        bytes[20..].copy_from_slice(&issuer.0);
-        Self {
-            issuer,
-            currency,
-            _bytes: bytes,
-        }
+impl Issue {
+    /// Create a new Issue
+    pub const fn new(currency: Currency, account: AccountID) -> Self {
+        Self { currency, account }
     }
 
-    pub fn as_bytes(&self) -> &[u8] {
-        &self._bytes
+    /// Check if this issue is native XRP
+    pub fn native(&self) -> bool {
+        self.currency.0 == XRP_CURRENCY.0 && self.account.0 == XRP_ACCOUNT.0
     }
 }
 
-/// Struct to represent an Issue of type MPT. Exists so that other structs can restrict type
-/// information to XRP in their declarations (this is not possible with just the `Issue` enum below).
+/// MPTIssue represents a Multi-Purpose Token issuance.
+/// This mirrors the C++ MPTIssue class.
+/// MptId is a type alias for UInt192 (24 bytes).
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 #[repr(C)]
-pub struct MptIssue {
+pub struct MPTIssue {
+    /// The MPT ID (24 bytes: 4 bytes sequence + 20 bytes account)
     mpt_id: MptId,
 }
 
-/// Represents an issue without a value, such as reading `Asset1` and `Asset2` in AMM ledger
-/// objects.
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-#[repr(C)]
-pub enum Issue {
-    XRP(XrpIssue),
-    IOU(IouIssue),
-    MPT(MptIssue),
+impl MPTIssue {
+    /// Create a new MPTIssue from an MPTID
+    pub const fn new(mpt_id: MptId) -> Self {
+        Self { mpt_id }
+    }
+
+    /// Get the MPTID
+    pub const fn mpt_id(&self) -> MptId {
+        self.mpt_id
+    }
+
+    /// Extract the issuer AccountID from the MPTID (bytes 4..24)
+    pub fn issuer(&self) -> AccountID {
+        let mut account_bytes = [0u8; 20];
+        account_bytes.copy_from_slice(&self.mpt_id.0[4..24]);
+        AccountID::from(account_bytes)
+    }
+
+    /// Extract the sequence number from the MPTID (bytes 0..4)
+    pub fn sequence(&self) -> u32 {
+        u32::from_be_bytes([
+            self.mpt_id.0[0],
+            self.mpt_id.0[1],
+            self.mpt_id.0[2],
+            self.mpt_id.0[3],
+        ])
+    }
+
+    /// Construct an MPTIssue from a sequence number and issuer AccountID
+    pub fn from(sequence_num: u32, issuer: AccountID) -> Self {
+        let mut bytes = [0u8; HASH192_SIZE];
```

Comment:
Thinking out loud -- but isn't it safer to have bytes in WASM that represent an MptId always have the "correct" byte encoding, which is 48 bytes?

(I'm thinking about the 20 blackhole account bytes that were added in the MptId in rippled to accomadate Issue -- see [here](https://github.com/ripple/xrpl-wasm-std/pull/64/files#diff-900ebcb59ebab996b6e9fb52fc2be501ec1597d398a6592d5df19799822b8ab0R137) in this PR)